packages/oidc-client/src/lib/oidc.api.ts

```diff
@@ -20,6 +22,159 @@ export const oidcApi = createApi({
   reducerPath: 'oidc',
   baseQuery: fetchBaseQuery(),
   endpoints: (builder) => ({
+    authorizeFetch: builder.mutation<AuthorizeSuccessResponse, { url: string }>({
+      queryFn: async ({ url }, api, _, baseQuery) => {
+        const { requestMiddleware, logger } = api.extra as Extras;
+
+        const request: FetchArgs = {
+          url,
+          method: 'POST',
+          credentials: 'include',
+          headers: {
+            Accept: 'application/json',
+          },
+        };
+
+        logger.debug('OIDC authorize API request', request);
+
+        const response = await initQuery(request, 'authorize')
+          .applyMiddleware(requestMiddleware)
+          .applyQuery(async (req: FetchArgs) => await baseQuery(req));
+
+        if (response.error) {
+          const responseError = response.error;
+
+          // If the details field is present in data, use it to create a more specific error response
+          if (
+            responseError.data &&
+            typeof responseError.data === 'object' &&
+            'details' in responseError.data &&
+            Array.isArray(responseError.data.details)
+          ) {
+            logger.debug('Error in authorize response', responseError);
+
+            const details = responseError.data.details[0] as {
+              code: string;
+              message: string;
+            };
+
+            response.error = {
+              status: responseError.status,
+              statusText: 'AUTHORIZE_ERROR',
+              data: {
+                error: details.code,
+                error_description: details.message,
+                type: 'auth_error',
+              },
+            } as FetchBaseQueryError;
+            return response;
+          }
+
+          logger.error('Error in OAuth configuration', responseError);
+
+          // Since this is likely a configuration issue, avoid providing a redirect URL
+          response.error = {
+            status: responseError.status,
+            statusText: 'CONFIGURATION_ERROR',
+            data: {
+              error: 'FETCH_ERROR',
+              error_description:
+                'Configuration error. Please check your OAuth configuration, like clientId or allowed redirect URLs.',
+              type: 'network_error',
+            },
+          } as FetchBaseQueryError;
```

Comment:
Same comment here. This isn't quite a FetchBaseQueryError. We might want to consider using `status: 'CUSTOM_ERROR'`